packages/test-utils/src/substrate/substrateNodeBase.ts

```diff
@@ -0,0 +1,339 @@
+        });
+      });
+    });
+  }
+
+  public async executeTransaction (
+    tx: SubmittableExtrinsic<'promise'>
+  ): Promise<string> {
+    const api = await this.api();
+    return new Promise((resolve, reject) => {
+      tx.send(({ status, dispatchError }) => {
+        // status would still be set, but in the case of error we can shortcut
+        // to just check it (so an error would indicate InBlock or Finalized)
+        if (dispatchError) {
+          if (dispatchError.isModule) {
+            // for module errors, we have the section indexed, lookup
+            const decoded = api.registry.findMetaError(dispatchError.asModule);
+            const { docs, name, section } = decoded;
+
+            reject(`${section}.${name}: ${docs.join(' ')}`);
+          } else {
+            // Other, CannotLookup, BadOrigin, no extra info
+            reject(dispatchError.toString());
+          }
+        }
+
+        if (status.isFinalized && !dispatchError) {
+          resolve(status.asFinalized.toString());
+        }
+      });
+    });
+  }
+
+  public async sudoExecuteTransaction (
+    tx: SubmittableExtrinsic<'promise'>
+  ): Promise<string> {
+    const api = await this.api();
+    const keyring = new Keyring({ type: 'sr25519' });
+    const sudoKey = keyring.addFromUri('//Alice');
+    const sudoCall = api.tx.sudo.sudo(tx);
+
+    return new Promise((resolve, reject) => {
+      sudoCall.signAndSend(
+        sudoKey,
+        { nonce: -1 },
+        ({ dispatchError, status }) => {
+        // status would still be set, but in the case of error we can shortcut
+        // to just check it (so an error would indicate InBlock or Finalized)
+          if (dispatchError) {
+            if (dispatchError.isModule) {
+            // for module errors, we have the section indexed, lookup
+              const decoded = api.registry.findMetaError(
+                dispatchError.asModule
+              );
+              const { docs, name, section } = decoded;
+
+              reject(`${section}.${name}: ${docs.join(' ')}`);
+            } else {
+            // Other, CannotLookup, BadOrigin, no extra info
+              reject(dispatchError.toString());
+            }
+          }
+
+          if (status.isFinalized && !dispatchError) {
+            resolve(status.asFinalized.toString());
+          }
+        }
+      );
+    });
+  }
+
+  protected static checkIfImageExists (image: string): boolean {
+    const result = execSync('docker images', { encoding: 'utf8' });
+
+    return result.includes(image);
+  }
+
+  protected static pullImage (opts: {
+    frocePull: boolean;
+    image: string;
+  }): void {
+    if (!this.checkIfImageExists(opts.image) || opts.frocePull) {
+      execSync(`docker pull ${opts.image}`, {
+        encoding: 'utf8'
+      });
+    }
+  }
+}
+
+async function createApiPromise (endpoint: string) {
+  return ApiPromise.create(
+    options({
+      provider: new WsProvider(endpoint) as any,
+      rpc: {
```

Comment:
Does that API have these hardcoded in it already? (`packages/api`). If it does can we instantiate the API from that package instead of having this hardcoded repeatedly?

Reply:
No, it does not.